src/upnp.cpp

```diff
@@ -996,9 +998,10 @@ void upnp::on_upnp_xml(error_code const& e
 		d.disabled = true;
 		return;
 	}
-	static std::string service_type;
-	service_type.swap(s.service_type);
-	d.service_namespace = service_type.c_str();
+	d.service_namespace = s.service_type;
```

Comment:
I wonder if it's ok to do this, maybe that swap was there for a good reason.
Also, it did a `.c_str()`, and now I see you've had to put `.c_str()` everywhere because you don't do it here it seems.